multi-agent/lib/mas/elements/nodes/common/context_binder.py

```
@@ -0,0 +1,97 @@
+            try:
+                tool.bind_context(session_id=session_id, agent_id=agent_id)
+            except Exception:
+                logger.warning(
+                    "Failed to bind context to tool %s",
+                    getattr(tool, "name", "unknown"),
+                    exc_info=True,
+                )
+
+
+def get_sandbox_wrapped_mcp_tools(
+    sandbox: Optional[Any],
```

Ruff (0.15.20)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `sandbox`

(ANN401)